java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/JniSinkWriterHandler.java

```diff
@@ -0,0 +1,237 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.risingwave.connector;
+
+import static com.risingwave.connector.SinkUtils.getConnectorName;
+import static io.grpc.Status.*;
+import static io.grpc.Status.INVALID_ARGUMENT;
+
+import com.risingwave.connector.api.TableSchema;
+import com.risingwave.connector.api.sink.*;
+import com.risingwave.connector.deserializer.StreamChunkDeserializer;
+import com.risingwave.java.binding.Binding;
+import com.risingwave.metrics.ConnectorNodeMetrics;
+import com.risingwave.metrics.MonitoredRowIterator;
+import com.risingwave.proto.ConnectorServiceProto;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JniSinkWriterHandler {
+    private static final Logger LOG = LoggerFactory.getLogger(JniSinkWriterHandler.class);
+    private SinkWriter sink;
+
+    private String connectorName;
+
+    private long sinkId;
+
+    private TableSchema tableSchema;
+
+    private boolean epochStarted;
+    private long currentEpoch;
+    private Long currentBatchId;
+
+    private Deserializer deserializer;
+
+    private long requestRxPtr;
+
+    private long responseTxPtr;
+
+    public boolean isInitialized() {
+        return sink != null;
+    }
+
+    public JniSinkWriterHandler(long requestRxPtr, long responseTxPtr) {
+        this.requestRxPtr = requestRxPtr;
+        this.responseTxPtr = responseTxPtr;
+    }
+
+    public static void runJniSinkWriterThread(long requestRxPtr, long responseTxPtr)
+            throws com.google.protobuf.InvalidProtocolBufferException {
+        // For jni.rs
+        java.lang.Thread.currentThread()
+                .setContextClassLoader(java.lang.ClassLoader.getSystemClassLoader());
+
+        JniSinkWriterHandler handler = new JniSinkWriterHandler(requestRxPtr, responseTxPtr);
+
+        byte[] requestBytes;
+        while ((requestBytes = Binding.recvSinkWriterRequestFromChannel(handler.requestRxPtr))
+                != null) {
+            var request = ConnectorServiceProto.SinkWriterStreamRequest.parseFrom(requestBytes);
+            if (!handler.onNext(request)) {
+                handler.cleanup();
+                return;
+            }
+        }
+        LOG.info("end of runJniSinkWriterThread");
+    }
+
+    public boolean onNext(ConnectorServiceProto.SinkWriterStreamRequest sinkTask) {
```

Comment:
Can we implement `StreamObserver` for a wrapper on the request receiver so that the current `SinkWriterObserver` can be reused without any change.